saba/main.py

```diff
@@ -439,6 +442,188 @@ def get_sampler(self, *args, **kwargs):
         """
         return SherpaMCMC(self, *args, **kwargs)
 
+    def remove_units(self, models, x, y, z=None, xbinsize=None, ybinsize=None, err=None, bkg=None, equivalencies=None):
+        """
+        This stripts data of there units so they can be fit.
+        """
+        #try to get a model, x, y, z, xbin, ybin, err, bkg tuple
+        #see how many models
+        try:
+            models._supports_unit_fitting
+            n_models = 1
+            models = [models]
+        except AttributeError:
+            n_models = len(models)
+
+        #try to figure out the is multiple datasets
+        #assume all data has same type?
+        try:
+            n_data = x.ndim
+            if z  is None:
+                assert x.ndim == y.ndim, AstropyUserWarning("x and y dimensions don't match")
```

Comment:
Why not just raise a ValueError here? Is there a use case where we should just proceed with a warning?